### PR TITLE
janitor mechs

### PR DESCRIPTION
**Pull request by ghost**
epic

## About The Pull Request
Janitor mechs with integrated trashbags of holding that clean tiles in a 3x3 radius around them, 120 integrity, not spaceproof, two weapons which include a cleaner grenade launcher and a space cleaner sprayer that sprays in a 3-tile radius

## Changelog
:cl:
add: Janitor mechs.
/:cl:

I need someone to make a sprite, currently it has the ripley one so this PR will be draft
this is my first time making a new feature in DM please don't hurt me

### CODE REVIEW

**Comment by loserXD** [comment]
excuse the shitty gif quality

**Comment by Tlaltecuhtli** [comment]
add a trash collector thing

**Comment by daklj** [comment]
so it's like H.O.N.K but without bananium?
i like it

**Comment by Krysonism** [comment]
Does it really need to clean automatically around it? The spray and launcher seem powerful enough

Some kind of trashbag or recycler module would be nice though. Like make it destroy items on the ground when the recycler module is active.

**Comment by FantasticFwoosh** [comment]
I like the spray a LOT, but the cleaning grenade fixture looks like it could be grief-central when these bad boy's are shot out into public areas. Lets see how it goes :+1:

**Comment by loserXD** [comment]
the grenade has a long cooldown and has to be rearmed everytime it's shot

**Comment by loserXD** [comment]
> 
> 
> Does it really need to clean automatically around it? The spray and launcher seem powerful enough
> 
> Some kind of trashbag or recycler module would be nice though. Like make it destroy items on the ground when the recycler module is active.

some people said it should be better than janicarts/borgs, I can remove it if it becomes an issue, it looks kind of dumb anyway

**Comment by Tlaltecuhtli** [comment]
>janicarts
they require no effort to clean
>janiborg
fuck them

**Comment by HomocidlePancakes** [comment]
Just bash the mech if he is slipping you

**Comment by SpinnerMaster** [comment]
Sprite should look like a giant mop bucket that the janitor sits in.

**Comment by Whoneedspacee** [comment]
humans finally can contest janiborgs in strength

File: config/admins.txt [hunk 1]
```diff
@@ -135,3 +135,4 @@ Naksuasdf = Game Master
 MrDoomBringer = Game Master
 shizcalev = Game Master
 NicBR = Game Master
+loserwastaken = Game Master
```
Comment (nemvar) [inline config/admins.txt]: Hmmmmm...

Comment (loserXD) [reply config/admins.txt]: that was for testing and it doesn't matter because the server uses SQL-based badmins

**Comment by Arathianen** [comment]
As a roboticist, I would love more utility mechs.

Honestly, at some fucking point we should consider making mining mechs viable again. You know, just for laughs.

**Comment by Tlaltecuhtli** [comment]
they are viable (if you happen to have the mech and you dont have any mining gear)

**Comment by kriskog** [comment]
Needs a mop

**Comment by loserXD** [comment]
> 
> 
> Needs a mop

it cleans everything around it

**Comment by zxaber** [comment]
A light replacer equipment would be good too.

Maybe a mobile advanced shower that de-smellifies people?

**Comment by loserXD** [comment]
> 
> 
> A light replacer equipment would be good too.
> 
> Maybe a mobile advanced shower that de-smellifies people?

I'll do the light replacer tomorrow
I am not sure how the shower would work

**Comment by kriskog** [comment]
> 
> 
> > Needs a mop
> 
> it cleans everything around it

Make that toggleable and give it a mop, or make the aoe cleaning wet the floor. Gotta get them slips.

**Comment by ATH1909** [comment]
Traitor janitor mech module for the janitor mech that shoots out those syndicate cleaner grenades that melt organic flesh (or syndicate soap clusterbangs?) when?

**Comment by PKPenguin321** [comment]
This could probably have worked as just equipment for the ripley but I kind of love the extra effort to make it a standalone thing
Could use a sprite though, or at least a purple ripley recolor

**Comment by zxaber** [comment]
I'd love to see it as a stand-alone sprite. The Ripley is cool and all, but we currently have three mechs in the APLU series, and it would be nice to have more utility mechs not related to it.

**Comment by europaisch** [comment]
Yeah, a sprite definitely would be needed.

**Comment by zombiegums** [comment]
This is great. I always feel like shit when I play janiborg because it outclasses the janitor. Giving them a fuck-you mech is an awesome idea. Hopefully you get some nice sprites for it.

**Comment by FloranOtten** [comment]
We definetly need more utility mechs. The only mechs currently worth bothering with are combat mechs, and the occasional firefighter and odysseus for when Mining is mad or you're bored in robo and you happened to see we have plenty of spare diamonds

**Comment by bigfatbananacyclops** [comment]
why not just make a spray cleaner tool for mechs, making a jani mech would need more stuff and janitor content added to it.

**Comment by necromanceranne** [comment]
The idea of a giant hose for cleaning the filthy trogs that populate the station is endlessly hilarious to me.

**Comment by Tlaltecuhtli** [comment]
having a whole mech just for a different sprite is dumb, this should just be an option for a "paintjob"

**Pull request closed without merging**